api/v1beta1/nodemodulestate_types.go

```diff
@@ -0,0 +1,87 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type ModuleNodeSpec struct {
+	Name            string                   `json:"name"`
+	Namespace       string                   `json:"namespace"`
+	ModuleLoader    ModuleLoaderSpec         `json:"moduleLoader"`
+	ImageRepoSecret *v1.LocalObjectReference `json:"imageRepoSecret,omitempty"`
+	KmomLoaded      bool                     `json:"kmodLoaded"`
```

Comment:
```suggestion
	KmodLoaded      bool                     `json:"kmodLoaded"`
```

Reply:
Do we need a full `ModuleLoaderSpec`? In my PoC I had
```go
type ModuleConfig struct {
	ContainerImage string `json:"containerImage"`

	//+optional
	InTreeModuleToRemove string `json:"inTreeModuleToRemove"`

	Modprobe ModprobeSpec `json:"modprobe"`
}
```

Reply:
ok, i can change it, we will update later if needed

Reply:
fixed